app/javascript/App/components/CoinIndex/index.js

```diff
@@ -1,46 +1,45 @@
 import React, { Component, Fragment } from 'react'
-import {
-  Layout,
-  Card,
-  Button,
-  Menu,
-  Dropdown,
-  Icon,
-  Table,
-  Pagination,
-} from 'antd'
-import styled from 'styled-components'
-import axios from 'axios'
+import { Table } from 'antd'
 import ColumnNames from './ColumnNames'
-
-const { Header, Footer, Content } = Layout
+import CurrencySelector from '../CurrencySelector'
+import SearchCoins from '../shared/SearchCoins'
+import API from '../../lib/localAPI'
 
 class CoinIndex extends Component {
   state = {
+    coins: [],
+    pagination: {
+      defaultPageSize: 10,
+      total: this.props.coinCount,
+    },
+    loading: false,
     currency: 'USD',
-    priceData: [],
   }
 
-  componentDidMount() {
-    axios
-      .get('/api/coins?limit=100')
-      .then((response) => {
-        this.setState({
-          priceData: response.data.payload,
-        })
-      })
-      .catch((error) => {
-        console.log(error)
-      })
+  handleTableChange = (pagination) => {
+    const pager = { ...this.state.pagination }
+    pager.current = pagination.current
+    this.setState({
+      pagination: pager,
+    })
+    this.fetchCoins({
+      per: pagination.pageSize,
+      page: pagination.current,
+    })
   }
 
-  getPageTotal() {
-    //todo: get total page count from header
-    return 160
+  fetchCoins = (params = {}) => {
+    this.setState({ loading: true })
+    API.get('/coinsnew', params).then((response) => {
+      this.setState({
+        loading: false,
+        coins: response,
+      })
+    })
   }
 
-  jumpPage(data) {
-    window.location = `/coinsnew?page=${data}`
+  componentDidMount() {
+    this.fetchCoins()
```

Comment:
moving away from server rendered pages, nice!

Reply:
Seems like it's easier to do so for Ant Table